packages/flutter/lib/src/widgets/text_editing_actions.dart

```diff
@@ -0,0 +1,584 @@
+class ControlArrowRightTextIntent extends TextEditingIntent {}
+class ControlCTextIntent extends TextEditingIntent {}
+class ControlShiftArrowLeftTextIntent extends TextEditingIntent {}
+class ControlShiftArrowRightTextIntent extends TextEditingIntent {}
+class DragSelectionEndTextIntent extends TextEditingIntent {}
+class DragSelectionStartTextIntent extends TextEditingIntent {}
+class DragSelectionUpdateTextIntent extends TextEditingIntent {}
+class ForcePressEndTextIntent extends TextEditingIntent {}
+class ForcePressStartTextIntent extends TextEditingIntent {}
+class EndTextIntent extends TextEditingIntent {}
+class HomeTextIntent extends TextEditingIntent {}
+class MetaArrowDownTextIntent extends TextEditingIntent {}
+class MetaArrowLeftTextIntent extends TextEditingIntent {}
+class MetaArrowRightTextIntent extends TextEditingIntent {}
+class MetaArrowUpTextIntent extends TextEditingIntent {}
+class MetaShiftArrowDownTextIntent extends TextEditingIntent {}
+class MetaShiftArrowLeftTextIntent extends TextEditingIntent {}
+class MetaShiftArrowRightTextIntent extends TextEditingIntent {}
+class MetaShiftArrowUpTextIntent extends TextEditingIntent {}
+class MetaCTextIntent extends TextEditingIntent {}
+class SingleLongTapEndTextIntent extends TextEditingIntent {}
+class SingleLongTapMoveUpdateTextIntent extends TextEditingIntent {}
+class SingleLongTapStartTextIntent extends TextEditingIntent {}
+class SingleTapCancelTextIntent extends TextEditingIntent {}
+class SingleTapUpTextIntent extends TextEditingIntent {
+  SingleTapUpTextIntent({
+    required this.details,
+  });
+
+  final TapUpDetails details;
+}
+class ShiftArrowDownTextIntent extends TextEditingIntent {}
+class ShiftArrowLeftTextIntent extends TextEditingIntent {}
+class ShiftArrowRightTextIntent extends TextEditingIntent {}
+class ShiftArrowUpTextIntent extends TextEditingIntent {}
+class ShiftEndTextIntent extends TextEditingIntent {}
+class ShiftHomeTextIntent extends TextEditingIntent {}
+
+/// The map of [Action]s that correspond to the default text editing behavior
+/// for Flutter on the current platform.
+///
+/// See also:
+///
+/// * [Actions], which can be used to override this behavior.
+class TextEditingActions extends StatelessWidget {
+  /// Creates an instance of TextEditingActions.
+  TextEditingActions({
+    Key? key,
+    Map<Type, Action<Intent>>? additionalActions,
+    required this.child,
+  }) : additionalActions = additionalActions ?? <Type, Action<Intent>>{},
+       super(key: key);
+
+  /// The child [Widget] of TextEditingActions.
+  final Widget child;
+
+  /// The actions to be merged with the default text editing actions.
+  ///
+  /// The default text editing actions will override any conflicting keys in
+  /// additionalActions. To override the default text editing actions, use an
+  /// [Actions] Widget in the tree below this Widget.
+  final Map<Type, Action<Intent>> additionalActions;
```

Comment:
I'm interested to know thoughts on this.  Instead of having this parameter, I could just use an additional Actions widget in widgets/app.dart.

Reply:
Same question for TextEditingShortcuts.additionalShortcuts as well.